### ISSUE TITLE
[BUG]  Address TextMeshPro dependency in Unity 2023

### ISSUE THREAD

**Issue by AMollis** [MixedRealityToolkit/MixedRealityToolkit-Unity#501]
## Context

1. TextMeshPro has been moved into [com.unity.ugui@2.0.0](mailto:com.unity.ugui@2.0.0)
2. Unity 2023 requires [com.unity.ugui@2.0.0](mailto:com.unity.ugui@2.0.0).    
3. [com.unity.ugui@2.0.0](mailto:com.unity.ugui@2.0.0) is not available in Unity 2021 or 2022
4. MRTK3 depends on com.unity.textmeshpro v3.0.6  (see [package.json](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/blob/main/org.mixedrealitytoolkit.uxcore/package.json) file)
5. MRTK3 needs to work across all LTS versions (Unity 2021, 2022, and eventually 2023)

Because of 1-4 we are unable to satisfy 5.   If we leave  [package.json](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/blob/main/org.mixedrealitytoolkit.uxcore/package.json) as is, Unity 2023 complains and gives "duplicate type definitions" errors. If we remove  com.unity.textmeshpro from  [package.json](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/blob/main/org.mixedrealitytoolkit.uxcore/package.json), Unity 2021 and Unity 2022 complains and gives "unknown type" errors. 

In the past, when Unity has moved classes or functionality to a new package, Unity has made "facade" packages and/or classes.  These "facades" redirect to the new package with the new implementations.  Can Unity do the same and create a facade for com.unity.textmeshpro in Unity 2023? 

Or can Unity recommend a solution for this problem?  So, we can maintain one MRTK3 version that works across 2021, 2022 and 2023.

**Comment by marlenaklein-msft** [comment]
Packages with dependency: accessibility, data, uxcomponents, uxcore

**Comment by AMollis** [comment]
@marlenaklein-msft please add the "package: *" labels for all impacted packages.

**Comment by shaynie** [comment]
Need to determine why textmesh pro still gets imported after removing our dependency.  @AMollis  please update.

**Comment by AMollis** [comment]
Updated description and title.   This requires a fix from Unity

**Comment by AMollis** [comment]
We are currently working on a solution for this.

If developers are blocked, they can:

1. Use the MRTK3 packages directly from the repository.  
2. Replace all references to com.unity.textmeshpro in package.json files with [com.unity.ugui@2.0.0](mailto:com.unity.ugui@2.0.0)
3. Update project manifest.json to remove reference to com.unity.textmeshpro

**Comment by anonymous2585** [comment]
@AMollis 
Something interesting happened in Unity 2023.2.7f1:
https://unity.com/releases/editor/whats-new/2023.2.7#release-notes
> TextMeshPro: Introduce a shim version of TextMesh Pro for packages that had a hard dependency on it. (UUM-58068)

I updated our project from 2023.1.16f1 to 2023.2.7f1 and got no issue at all, no need to edit MRTK's `package.json`, nothing to do.

**Comment by AMollis** [comment]
Thanks for verifying @anonymous2585 ...

We've been working on fix with Unity here.  I'm glad it's working.   Resolving issue as fixed externally
